lib/resourceapply/security_test.go

```diff
@@ -35,6 +36,16 @@ var restrictedv2 = securityv1.SecurityContextConstraints{
 	SeccompProfiles:          []string{"runtime/default"},
 }
 
+func defaulted(get func() *securityv1.SecurityContextConstraints) func() *securityv1.SecurityContextConstraints {
+	return func() *securityv1.SecurityContextConstraints {
+		scc := get()
+		if scc.AllowPrivilegeEscalation == nil {
+			scc.AllowPrivilegeEscalation = pointer.Bool(true)
+		}
+		return scc
+	}
+}
+
```

Comment:
Any reason you want to use a separate function to set `scc.AllowPrivilegeEscalation = pointer.Bool(true)` vs you use set scc.AllowPrivilegeEscalation = true in `var restrictedv2` and only for test case `name: "no modification when existing differs but there's release.openshift.io/create-only: true annotation",` set it to nil.

Reply:
Two reasons. I wanted to keep restrictedv2 as a true version of the restricted-v2 SCC from 4.14 payload, but more importantly, I prefer surprising behavior (like defaulting) to be as explicit as possible. I could set up `var restrictedv2` in a way that works but it woudl be suprising if you e.g. removed that member from test data from whatever reason and all tests started to fail. Done this way, testcases hint about the case that the existing state will be defaulted, the manifest will not, and the expected state is defaulted. E.g. this case shows it nicely:

```
name:     "no modification when existing is equal to required",
existing: defaulted(restrictedv2.DeepCopy),
required: restrictedv2.DeepCopy,
expected: defaulted(restrictedv2.DeepCopy),
...
```